src/cli/clear.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+// implementation of `sourcecred clear`
+
+import path from "path";
+import fs from "fs";
+import rimraf from "rimraf";
+
+import dedent from "../util/dedent";
+import {type Command} from "./command";
+import * as Common from "./common";
+
+function usage(print: (string) => void): void {
+  print(
+    dedent`\
+    usage: sourcecred clear --all
+           sourcecred clear --cache
+           sourcecred clear --help
+
+    Remove the SOURCECRED_DIRECTORY, i.e. the directory where data, caches,
+    registries, etc. owned by SourceCred are stored.
+
+    Arguments:
+        --all
+            remove entire SOURCECRED_DIRECTORY
+
+        --cache
+            remove only the SourcCred cache directory
+
+        --help
+            Show this help message and exit, as 'sourcecred help clear'.
+
+    Environment Variables:
+        SOURCECRED_DIRECTORY
+            Directory owned by SourceCred, in which data, caches,
+            registries, etc. are stored. Optional: defaults to a
+            directory 'sourcecred' under your OS's temporary directory;
+            namely:
+                ${Common.defaultSourcecredDirectory()}
+
+    `.trimRight()
+  );
+}
+
+function die(std, message) {
+  std.err("fatal: " + message);
+  std.err("fatal: run 'sourcecred help clear' for help");
+  return 1;
+}
+
+export function makeClear(removeDir: (string) => void): Command {
+  return async function clear(args, std) {
+    if (args.length === 0) {
+      return die(std, "no arguments provided");
+    }
+
+    if (args.length > 1) {
+      return die(
+        std,
+        `expected 1 argument but recieved: ${args.length} arguments`
+      );
+    }
+
+    switch (args[0]) {
+      case "--help":
+        usage(std.out);
+        return 0;
+      case "--all":
+        return removeDir(Common.sourcecredDirectory());
+      case "--cache":
+        const cacheDir = path.join(Common.sourcecredDirectory(), "cache");
+        return removeDir(cacheDir);
+      default:
+        return die(std, `'${args[0]}' is not a valid argument`);
+    }
+  };
+}
+
+//how do we handle callbacks
```

Comment:
We generally prefer to use promises and async rather than callbacks. The other commands import "fs-extra" rather than "fs". 

https://www.npmjs.com/package/fs-extra